starknet_devnet/starknet_wrapper.py

```diff
@@ -329,14 +334,19 @@ async def call(self, transaction: InvokeFunction):
 
         return { "result": adapted_result }
 
-    async def __register_new_contracts(self, execution_info: TransactionExecutionInfo):
-        for internal_call in execution_info.call_info.internal_calls:
+
+
+    async def __register_new_contracts(self, internal_calls: List[Union[FunctionInvocation, CallInfo]]):
```

Comment:
Changed from execution_info to internal_calls to enable recursive calls (had to enable recursion to support delegate calls).

Had to use Union of types because `internal_calls` isn't of the same type for Deploy and Invoke transactions - one difference that affected Devnet is internal_call.class_hash: `int` for one type, `bytes` for the other..